nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongoBulkOperations.java

```diff
@@ -0,0 +1,304 @@
+                    final UpdateOptions options = parseUpdateOptions(updateSpec);
+                    writeModel = new UpdateOneModel<>((BsonDocument) updateSpec.get("filter"), (BsonDocument) updateSpec.get("update"), options);
+                } else if ("updateMany".equals(updateType)) {
+                    final UpdateOptions options = parseUpdateOptions(updateSpec);
+                    writeModel = new UpdateManyModel<>((BsonDocument) updateSpec.get("filter"), (BsonDocument) updateSpec.get("update"), options);
+                } else if ("replaceOne".equals(updateType)) {
+                    final ReplaceOptions options = parseReplaceOptions(updateSpec);
+                    writeModel = new ReplaceOneModel<>((BsonDocument) updateSpec.get("filter"),
+                            toBsonDocument((BsonDocument) updateSpec.get("replacement")), options);
+                } else if ("deleteOne".equals(updateType)) {
+                    final DeleteOptions options = parseDeleteOptions(updateSpec);
+                    writeModel = new DeleteOneModel<>((BsonDocument) updateSpec.get("filter"), options);
+                } else  if ("deleteMany".equals(updateType)) {
+                    final DeleteOptions options = parseDeleteOptions(updateSpec);
+                    writeModel = new DeleteManyModel<>((BsonDocument) updateSpec.get("filter"), options);
+                } else {
+                    logger.error("Invalid bulk-update: invalid update type {}", updateType);
+                    session.transfer(flowFile, REL_FAILURE);
+                    context.yield();
+                    return;
+                }
+                updateModels.add(writeModel);
+            }
+
+            if (context.getProperty(USE_TRANSACTION).asBoolean()) {
+                clientSession = clientService.startSession();
+                clientSession.startTransaction();
+                // now run this w/in a transaction
+                collection.bulkWrite(clientSession, updateModels, (new BulkWriteOptions().ordered(context.getProperty(ORDERED).asBoolean())));
+            } else {
+                collection.bulkWrite(updateModels, (new BulkWriteOptions().ordered(context.getProperty(ORDERED).asBoolean())));
+            }
+            logger.info("bulk-updated {} into MongoDB", flowFile);
+            // (could also return the result again as JSON - mostly not needed afaik)
+
+            session.getProvenanceReporter().send(flowFile, getURI(context));
+            session.transfer(flowFile, REL_SUCCESS);
+
+            if (clientSession != null) {
+                if (clientSession.hasActiveTransaction()) {
+                    clientSession.commitTransaction();
+                }
+                clientSession.close();
+            }
+        } catch (Exception e) {
+            logger.error("Failed to bulk-update {} into MongoDB", flowFile, e);
+            session.transfer(flowFile, REL_FAILURE);
+            context.yield();
```

Comment:
Is there a reason for this yield?

Reply:
Same as PutMongo - and I think it makes sense to back off on error